File: NumberPossibilities/Scala/NumberPossibilities.scala

object NumberPossibilities
{
	def main(args: Array[String])
	{
		val start=System.nanoTime()
		//digit count starts from 0.
		val digitCount:Int=9
		var superNumbers=new Array[Int](digitCount)
		val possibilityCount : Int=math.pow(digitCount,digitCount).toInt
		for(i<-0 until possibilityCount-1)
		{
			incrementArray(superNumbers,superNumbers.length-1)
		}
		print("Max Digits:");
		superNumbers.foreach(print)
	
		val endTime :Double=System.nanoTime().toDouble
		val seconds :Double = (endTime-start) / 1000000000.0d

		println("\nTotal Seconds:"+seconds)
	}

	def incrementArray(possibilities: Array[Int],digitIndex:Int)
	{
		var tmp:Int=0
		var digitCount:Int = possibilities.length
		tmp=possibilities(digitIndex)+1
		if(tmp%digitCount==0)
		{
			possibilities(digitIndex)=0
			incrementArray(possibilities,digitIndex-1)
		}
		else
			possibilities(digitIndex)=tmp
	}
}

File: NumberPossibilities/Java/NumberPossibilities.java
public class NumberPossibilities
{
    public static void main(String[] args)
    {
        long startTime = System.currentTimeMillis();
        int[] superNumbers = new int[9];
        int j=9;
       // for (int j = 2; j <= 9; j++)
        {
            superNumbers = new int[j];
            int possibilityCount = (int)Math.pow(j, j);
            for (int i = 0; i < possibilityCount - 1; i++)
            {
                incrementArray(superNumbers, superNumbers.length - 1);
            }
        }
        System.out.print("Max Digit:");
            for (int digit : superNumbers)
                System.out.print(digit);
        long stopTime = System.currentTimeMillis();
        long elapsedTime = (stopTime - startTime)/1000;
        System.out.print("\nTime Taken(Seconds):"+elapsedTime);
    }
    public static void incrementArray(int[] possibilities, int digitIndex)
    {
        int tmp = 0;
        int digitCount = possibilities.length;
        tmp = possibilities[digitIndex] + 1;
        if (tmp % digitCount == 0)
        {
            possibilities[digitIndex] = 0;
            incrementArray(possibilities, digitIndex - 1);
        }
        else
            possibilities[digitIndex] = tmp;
    }
}